core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -216,12 +249,14 @@ private FileInStream getExternalFileInStream(long pos) throws IOException {
    * Reads a page from external storage which contains the position specified. Note that this makes
    * a copy of the page.
    *
+   * This method is synchronized to ensure thread safety for positioned reads.
```

Comment:
not clear to me what are we guarding. in addition, do we only provide thread-safety for positioned reads not normal reads?

Reply:
Yeah, only for positioned reads (PositionedReadable). The read API (InputStream) is not thread safe.

Reply:
I explained a bit more in the comment.